### DISCUSSION TITLE
Frequent puma crashes "rb_vm_insn_addr2insn: invalid insn address"

### DISCUSSION THREAD

**Discussion by turion** [mastodon/mastodon#21311] [Q&A] [answered]
Since updating to 4.0.2, I'm experiencing frequent 502s which seem to be caused by crashes of the puma threads:

```
Nov 21 15:25:18 manuelbaerenz puma[1006220]: /nix/store/1m8li3dkfj33kk4xhx6bhjbqi6y9bq6x-mastodon-gems-4.0.2/lib/ruby/gems/3.0.0/gems/activesupport-6.1.7/lib/active_support/json/encoding.rb:58: [BUG] rb_vm_insn_addr2insn: invalid insn address: 0x00007f0646b56e20
```
(See attached log, 10k lines)
[mastodon.log](https://github.com/mastodon/mastodon/files/10057376/mastodon.log)

Any idea what could be the cause of this? I'm running mastodon on dedicated hardware with NixOS.

**Comment by nightpool** [comment, marked as answer]
This is a crash in the Ruby interpreter, you'll have to take it up with the Nix project or the Ruby project, there's not much we can do about it.

**Reply by turion** [reply]
I reported as https://bugs.ruby-lang.org/issues/19140

**Comment by mjankowski** [comment]
Looks like the linked ruby bug tracker issue here was closed as being resolved by machine reboot ... going to close this on assumption there's nothing further mastodon-related to track here.
